packages/core/src/modules/action-menu/messages/MenuMessage.ts

```diff
@@ -0,0 +1,53 @@
+import type { ActionMenuOptionOptions } from '../models'
+
+import { Type } from 'class-transformer'
+import { IsInstance, IsOptional, IsString } from 'class-validator'
+
+import { AgentMessage } from '../../../agent/AgentMessage'
+import { IsValidMessageType, parseMessageType } from '../../../utils/messageType'
+import { ActionMenuOption } from '../models'
+
+export interface MenuMessageOptions {
+  id?: string
+  title: string
+  description: string
+  errormsg?: string
+  options: ActionMenuOptionOptions[]
+  threadId?: string
+}
+
+export class MenuMessage extends AgentMessage {
+  public constructor(options: MenuMessageOptions) {
+    super()
+
+    if (options) {
+      this.id = options.id ?? this.generateId()
+      this.title = options.title
+      this.description = options.description
+      this.options = options.options.map((p) => new ActionMenuOption(p))
+      if (options.threadId) {
+        this.setThread({
+          threadId: options.threadId,
+        })
+      }
+    }
+  }
+
+  @IsValidMessageType(MenuMessage.type)
+  public readonly type = MenuMessage.type.messageTypeUri
+  public static readonly type = parseMessageType('https://didcomm.org/action-menu/1.0/menu')
+
+  @IsString()
+  public title!: string
+
+  @IsString()
+  public description!: string
+
+  @IsString()
+  @IsOptional()
+  public errormsg?: string
```

Comment:
Maybe we can use an `Expose` here with different name and call it `errorMsg` or `errorMessage` in AFJ? Seems more aligned with the rest